x-pack/plugins/apm/public/components/shared/environment_filter/index.tsx

```diff
@@ -28,15 +28,23 @@ function updateEnvironmentUrl(
 }
 
 export function ApmEnvironmentFilter() {
-  const { environment, environments, status } = useEnvironmentsContext();
+  const { environment, environments, status, rangeFrom, rangeTo, serviceName } =
+    useEnvironmentsContext();
   const history = useHistory();
   const location = useLocation();
 
+  if (!rangeFrom || !rangeTo) {
+    return null;
+  }
+
   return (
     <EnvironmentSelect
       status={status}
       environment={environment}
       availableEnvironments={environments}
+      serviceName={serviceName}
+      rangeFrom={rangeFrom as string}
+      rangeTo={rangeTo as string}
```

Comment:
I think you don't need to cast to string anymore, right?

```suggestion
      rangeFrom={rangeFrom}
      rangeTo={rangeTo}
```

Reply:
True! I forgot about this. Thanks!